rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_storage.cpp

```diff
@@ -35,41 +36,43 @@
 
 namespace rosbag2_storage_plugins
 {
-
-SqliteStorage::SqliteStorage()
-: database_(),
-  write_statement_(nullptr),
-  read_statement_(nullptr),
-  message_result_(nullptr),
-  current_message_row_(nullptr, SqliteStatementWrapper::QueryResult<>::Iterator::POSITION_END)
-{}
-
 void SqliteStorage::open(
   const std::string & uri, rosbag2_storage::storage_interfaces::IOFlag io_flag)
 {
   auto metadata = is_read_only(io_flag) ?
     load_metadata(uri) :
     std::unique_ptr<rosbag2_storage::BagMetadata>();
 
+  std::string database_name;
+
   if (metadata) {
     if (metadata->relative_file_paths.empty()) {
       throw std::runtime_error(
               "Failed to read from bag '" + uri + "': Missing database file path in metadata");
     }
 
-    database_name_ = metadata->relative_file_paths[0];
+    database_name = metadata->relative_file_paths[0];
   } else {
     if (is_read_only(io_flag)) {
       throw std::runtime_error("Failed to read from bag '" + uri + "': No metadata found.");
     }
 
-    database_name_ = rosbag2_storage::FilesystemHelper::get_folder_name(uri) + ".db3";
+    std::stringstream ss;
+    ss << rosbag2_storage::FilesystemHelper::get_folder_name(uri);
```

Comment:
Maybe the way you've written it is better but I'd consider 

```
ss << rosbag2_storage::FilesystemHelper::get_folder_name(uri) << database_file_counter_ ? "_" : "" << ".db3";
```

But maybe that's a little overclever code golf